pep-0577.rst

```diff
@@ -1,14 +1,41 @@
 PEP: 577
 Title: Augmented Assignment Expressions
 Author: Nick Coghlan <ncoghlan@gmail.com>
-Status: Draft
+Status: Withdrawn
 Type: Standards Track
 Content-Type: text/x-rst
 Created: 14-May-2018
 Python-Version: 3.8
 Post-History: 22-May-2018
 
 
+PEP Withdrawal
+==============
+
+While working on this PEP, I realised that it didn't really address what was
+actually bothering me about PEP 572's proposed scoping rules for previously
+unreferenced assignment targets, and also had some significant undesirable
+consequences (most notably, allowing ``>>=` and ``<<=`` as inline augmented
+assignment operators that meant something entirely different from the ``>=``
+and ``<=`` comparison operators).
+
+I also realised that even without dedicated syntax of their own, PEP 572 allows
+inline augmented assignments to be written using the ``operator`` module::
+
+    from operator import iadd
+    if (target := iadd(target, value)) < limit:
```

Comment:
If someone sent me a code review containing that in a PEP 572 world I'd reject it. In most cases it's clearer written as `target := target + value` -- and in those cases where it makes a difference (e.g. because you don't know if `__iadd__` for this particular object returns `self` or a new object) you'd be better off using the non-inline syntax. So I don't see how this argument has any bearing on the debate about inline operators -- they don't add new functionality you can't write otherwise, they're meant to shorten code when it's straightforward, and if such a consideration was ever a concern the code is no longer straightforward.

Reply:
This doesn't really have any bearing on PEP 572, except insofar as it supports the argument that syntactic support for inline augmented assignment isn't necessary.